Volha_Ivanova/Module_1_Java/Classes_and_Objects/src/com/epam/cdp/volha_ivanova/task1/sweets/Gift.java

```diff
@@ -0,0 +1,64 @@
+package com.epam.cdp.volha_ivanova.task1.sweets;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+
+import com.epam.cdp.volha_ivanova.task1.sweets.candies.chocolateCandies.ChocolateCandies;
+import com.epam.cdp.volha_ivanova.task1.sweets.candies.lollipop.Lollipop;
+import com.epam.cdp.volha_ivanova.task1.sweets.chocolate.Chocolate;
+import com.epam.cdp.volha_ivanova.task1.sweets.waffles.Waffles;
+
+public class Gift {	
+	
+	int giftWeight;			
+	Scanner scanner = new Scanner(System.in);
+	Waffles waffle = new Waffles("Waffles", 10, scanner.nextInt());	
+	Chocolate chocolate1 = new Chocolate("White chocolate", "white", 100, scanner.nextInt());	
+	Chocolate chocolate2 = new Chocolate("Dark chocolate", "dark", 100, scanner.nextInt());	
+	Chocolate chocolate3 = new Chocolate("Milk chocolate", "Milk", 100, scanner.nextInt());	
+	ChocolateCandies chocoCandy = new ChocolateCandies("Chocolate candies", 30, scanner.nextInt());	
+	Lollipop lollipop = new Lollipop("Lollipop", 20, scanner.nextInt());				
+	
+	public void evaluateGiftWeight() {		
+		giftWeight = waffle.getWaffleCount()*waffle.getWeight() + chocolate1.getChocolateCount()*chocolate1.getWeight() + chocolate2.getChocolateCount()*chocolate2.getWeight()+chocolate3.getChocolateCount()*chocolate3.getWeight() + chocoCandy.getChocoCandiesCount()*chocoCandy.getWeight()+ lollipop.getLollipopCount()*lollipop.getWeight();
+		System.out.println("Weight of the present is " + giftWeight + " gramm");
+	}
+	public void sortCandies() {
+		HashMap<Integer, String> sweetsSort = new HashMap<Integer, String>();
```

Comment:
Every time when you want to sort or find candies you create new object. It's not optimal solution.
you should have the map as field.
sort and find method should perform only sorting or finding stuff